src/torrent_summary.rs

```diff
@@ -146,4 +167,85 @@ impl TorrentSummary {
 
     table
   }
+
+  pub(crate) fn write_json(&self, env: &mut Env) -> Result<()> {
```

Comment:
Duplicating all of this logic is not great. It'd probably be best to unify if as much as possible.

Reply:
I think it's probably okay not to unify it. The table format has some inherent limitations, in particular, it's not possible to represent more than two levels of nesting, so it's okay if the JSON output diverges somewhat.